js/interface.js

```diff
@@ -555,6 +571,217 @@ function uploadFiles(files) {
+        '>' + foldersForMoving['navStack'][i].name + '</a>' +
+        '</span>';
+    }
+    
+    $('#move-modal .current-folder-title').html(breadcrumbsPath);
+  }
+}
+
+//Create array tree form search result
+function createTreeFoldersArr(arr) {
+  var currentArr = arr;
+  
+  function transformToTree(arr) {
+    var nodes = {};
+    return arr.filter(function(obj) {
+      var id = obj["id"],
+        parentId = obj["parentId"];
+      
+      nodes[id] = _.defaults(obj, nodes[id], { children: [] });
+      parentId && (nodes[parentId] = (nodes[parentId] || { children: [] }))["children"].push(obj);
+      
+      return !parentId;
+    });
+  }
+  return transformToTree(currentArr);
+}
+
+//Create and open move modal
+function openMovePopup() {
+  foldersForMoving['navStack'] = [];
+  selectedItemsForMoving = $('.file-row.active');
+  var objData = {
+    items: selectedItemsForMoving,
+    itemsLength: selectedItemsForMoving.length
+  };
+  $('.file-manager-wrapper').append(templates.moveModal(objData));
+  $('#move-modal').modal('show');
+  organizationsList.forEach(addOrganizationsToSelect);
+  var $moveSelectPlaces = $('#move-model-organizations-select').find('option').first();
+  var id = $moveSelectPlaces.attr('data-id');
+  var type = $moveSelectPlaces.attr('data-type');
+  var name = $moveSelectPlaces.val();
+  addPathToStack(id, name, null, null, null, type);
+  this.searchFolders(id, type);
+}
+
+//Function for opening children folders
+function openChildrenFolders(folderId) {
+  var folder;
+  var folders = [];
+  var id = $('#move-model-organizations-select').find(':selected').attr('data-id');
+  
+  foldersForMoving[id].forEach(function (currChild) {
+    if (!!searchTree(currChild, folderId)) {
+      folder = searchTree(currChild, folderId);
+    }
+    folder = searchTree(currChild, folderId);
+  });
+  
+  folders = folder.children;
+  $('.move-modal-list').html('');
+  addFoldersToMoveModal(folders);
+  createMoveModalBreadCrumbsPaths();
+}
+
+// Call method for searching folders for selected app/organization
+function searchFolders(id, type) {
+  var res;
+  $('.move-modal-list').html('');
+  if (checkExistsMovingFolders(id)) {
+    res = checkExistsMovingFolders(id);
+    addFoldersToMoveModal(res);
+    createMoveModalBreadCrumbsPaths();
+  } else {
+    var filterFolders = function(folders) {
+      return true
+    };
+    var body = {};
+    if (type === "organizationId") {
+      body.organizationId = id;
+      filterFolders = function(folder) {
+        return (!(folder.appId || folder.parentFolderId) && folder.type === 'folder');
+      };
+    } else if (type === "appId") {
+      body.appId = id;
+      filterFolders = function(folder) {
+        return (!(folder.parentFolderId) && folder.type === 'folder');
+      };
+    }
+    $('.move-modal-loading').addClass('visible');
+    Fliplet.Media.Folders.search(body).then(function(response) {
+      res = response.filter(filterFolders);
```

Comment:
why is `res` required to be a variable? you could just mutate `response`?

Reply:
fixed